src/net.cpp

```diff
@@ -643,8 +643,6 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
 
 void CNode::CloseSocketDisconnect(CConnman* connman)
 {
-    AssertLockHeld(connman->m_nodes_mutex);
```

Comment:
801545adf56f8fe3f01ee7df48a2e54a0d6206f2: why is it impossible?

Reply:
Sorry, wrong wording, you don't need this lock at all